pkg/odo/genericclioptions/errors.go

```diff
@@ -1,11 +1,39 @@
 package genericclioptions
 
+import (
+	"fmt"
+
+	"github.com/redhat-developer/odo/pkg/devfile/location"
+	"github.com/redhat-developer/odo/pkg/testingutil/filesystem"
+)
+
 var _ error = NoDevfileError{}
 
-type NoDevfileError struct{}
+type NoDevfileError struct {
+	context string
+}
+
+func NewNoDevfileError(context string) NoDevfileError {
+	return NoDevfileError{
+		context: context,
+	}
+}
 
 func (o NoDevfileError) Error() string {
-	return "no devfile found"
+	message := `The current directory does not represent an odo component. 
+To get started:%s
+  * Open this folder in your favorite IDE and start editing, your changes will be reflected directly on the cluster.
+Visit https://odo.dev for more information.`
+
+	if isEmpty, _ := location.DirIsEmpty(filesystem.DefaultFs{}, o.context); isEmpty {
+		message = fmt.Sprintf(message, `
+  * Use "odo init" to initialize an odo component in the folder.
+  * Use "odo dev" to deploy it on cluster.`)
+	} else {
```

Comment:
For some reason, this always prints the second message, it does not seem to be detecting empty directory correctly.

<details>
<summary>O/p from odo delete component</summary>

```sh
➜  mkcd /tmp/101       
➜  odo delete component
 ✗  The current directory does not represent an odo component. 
To get started:
  * Use "odo dev" to initialize an odo component for this folder and deploy it on cluster.
  * Open this folder in your favorite IDE and start editing, your changes will be reflected directly on the cluster.
Visit https://odo.dev for more information.

```
</details

Reply:
Fixed, thanks